src/main/java/tools/jackson/dataformat/xml/deser/XmlDeserializationContext.java

```diff
@@ -68,7 +68,7 @@ public String extractScalarFromObject(JsonParser p, ValueDeserializer<?> deser,
             final String propName = p.currentName();
             JsonToken t = p.nextToken();
             if (t == JsonToken.VALUE_STRING) {
-                if (propName.equals("")) {
+                if (FromXmlParser.DEFAULT_TEXT_PROPERTY.equals(propName)) {
```

Comment:
I think this needs to actually consider `FromXmlParser._cfgNameForTextElement`, since that may be configured different from default. Comment above hints at that.

Will change to do that.